components/authz-service/server/v2/rules_property_test.go

```diff
@@ -18,17 +21,511 @@ import (
 )
 
 const (
-	idRegex = "^[a-z0-9-]{1,64}$"
+	idRegex        = "^[a-z0-9-]{1,64}$"
+	conditionLimit = 10 // avoid the grpc limit of 4194304
 )
 
+type projectAndRuleReq struct {
+	api.CreateProjectReq
+	rules []api.CreateRuleReq
+}
+
 func TestCreateRuleProperties(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	testFW := testhelpers.NewTestFramework(t, ctx)
-	cl := testFW.Projects
-	testDB := testFW.TestDB
-	seed := testFW.Seed
+	ctx := context.Background()
+	cl, testDB, _, _, seed := testhelpers.SetupProjectsAndRulesWithDB(t)
+	properties := getGopterParams(seed)
+	createRuleReqGen, createProjectReqGen, createProjectAndRuleGen := getGenerators()
+
+	properties.Property("creates a staged rule and confirms it is staged",
```

Comment:
💭 Is that a _property_? I'd propose something like
> all created rules are staged

Similarly below,
> applying a staged rule makes it applied

It's a bit silly, but describing _what the test does_ does read a bit odd to me 🤔

Reply:
See if you think the new set of names capture the spirit more correctly.